src/persist/src/client.rs

```diff
@@ -561,6 +561,84 @@ impl AtomicWriteBuilder<'_> {
     }
 }
 
+/// A handle for atomically allowing compaction on multiple streams, to the same frontier.
+#[derive(Debug, Clone)]
+pub struct AtomicCompactionHandle {
```

Comment:
as mentioned offline, I have a reasonably strong preference for folding these changes into MultiWriteHandle instead. we'll have to go that direction anyone when we decide to deal with our indefinite error problem

Reply:
:+1: